pkg/proxy/proxy.go

```diff
@@ -157,14 +157,14 @@ func (p *Proxy) Run(ctx context.Context) error {
 
 	for nAttempts < maxConnectAttempts {
 		session, err := p.createSession(ctx)
-		*p.cfg.DeviceToken = session.DeviceToken
```

Comment:
this could cause a segfault if we received an error back from the server. luckily we never deployed this version so no harm done

Reply:
Good catch